plugins/workload-management/src/test/java/org/opensearch/plugin/wlm/QueryGroupTestUtils.java

```diff
@@ -0,0 +1,146 @@
+        .build();
+
+    public static List<QueryGroup> queryGroupList() {
+        List<QueryGroup> list = new ArrayList<>();
+        list.add(queryGroupOne);
+        list.add(queryGroupTwo);
+        return list;
+    }
+
+    public static ClusterState clusterState() {
+        final Metadata metadata = Metadata.builder().queryGroups(Map.of(_ID_ONE, queryGroupOne, _ID_TWO, queryGroupTwo)).build();
+        return ClusterState.builder(new ClusterName("_name")).metadata(metadata).build();
+    }
+
+    public static Set<Setting<?>> clusterSettingsSet() {
+        Set<Setting<?>> set = new HashSet<>(ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+        set.add(QueryGroupPersistenceService.MAX_QUERY_GROUP_COUNT);
+        assertFalse(ClusterSettings.BUILT_IN_CLUSTER_SETTINGS.contains(QueryGroupPersistenceService.MAX_QUERY_GROUP_COUNT));
+        return set;
+    }
+
+    public static Settings settings() {
+        return Settings.builder().build();
+    }
+
+    public static ClusterSettings clusterSettings() {
+        return new ClusterSettings(settings(), clusterSettingsSet());
+    }
+
+    public static QueryGroupPersistenceService queryGroupPersistenceService() {
+        ClusterApplierService clusterApplierService = new ClusterApplierService(
+            "name",
+            settings(),
+            clusterSettings(),
+            mock(ThreadPool.class)
+        );
+        clusterApplierService.setInitialState(clusterState());
+        ClusterService clusterService = new ClusterService(
+            settings(),
+            clusterSettings(),
+            mock(ClusterManagerService.class),
+            clusterApplierService
+        );
+        return new QueryGroupPersistenceService(clusterService, settings(), clusterSettings());
+    }
+
+    public static List<Object> preparePersistenceServiceSetup(Map<String, QueryGroup> queryGroups) {
+        Metadata metadata = Metadata.builder().queryGroups(queryGroups).build();
+        Settings settings = Settings.builder().build();
+        ClusterState clusterState = ClusterState.builder(new ClusterName("_name")).metadata(metadata).build();
+        ClusterSettings clusterSettings = new ClusterSettings(settings, clusterSettingsSet());
+        ClusterApplierService clusterApplierService = new ClusterApplierService(
+            "name",
+            settings(),
+            clusterSettings(),
+            mock(ThreadPool.class)
+        );
+        clusterApplierService.setInitialState(clusterState);
+        ClusterService clusterService = new ClusterService(
+            settings(),
+            clusterSettings(),
+            mock(ClusterManagerService.class),
+            clusterApplierService
+        );
+        QueryGroupPersistenceService queryGroupPersistenceService = new QueryGroupPersistenceService(
+            clusterService,
+            settings,
+            clusterSettings
+        );
+        return List.of(queryGroupPersistenceService, clusterState);
```

Comment:
nitpick - return a Tuple<QueryGroupPersistenceService, ClusterState> instead of `List<Object>`.